airflow/providers/google/cloud/sensors/pubsub.py

```diff
@@ -170,22 +171,36 @@ def execute(self, context: Context) -> None:
                     subscription=self.subscription,
                     max_messages=self.max_messages,
                     ack_messages=self.ack_messages,
-                    messages_callback=self.messages_callback,
                     poke_interval=self.poke_interval,
                     gcp_conn_id=self.gcp_conn_id,
                     impersonation_chain=self.impersonation_chain,
                 ),
                 method_name="execute_complete",
             )
 
-    def execute_complete(self, context: dict[str, Any], event: dict[str, str | list[str]]) -> str | list[str]:
-        """Return immediately and relies on trigger to throw a success event. Callback for the trigger."""
+    def execute_complete(self, context: Context, event: dict[str, str | list[str]]) -> Any:
+        """If messages_callback is provided, execute it; otherwise, return immediately with trigger event message."""
         if event["status"] == "success":
             self.log.info("Sensor pulls messages: %s", event["message"])
+            if self.messages_callback:
+                received_message_fmt = self._convert_to_received_message(event["message"])
+                message_callback_response = self.messages_callback(received_message_fmt, context)
+
+                return message_callback_response
+
             return event["message"]
         self.log.info("Sensor failed: %s", event["message"])
         raise AirflowException(event["message"])
 
+    def _convert_to_received_message(self, messages: Any) -> list[Any]:
```

Comment:
```suggestion
    def _convert_to_received_message(self, messages: Any) -> list[ReceivedMessage]:
```

Sorry for the wrong suggestion previously. 🤦‍♂️ I was a bit confused

Reply:
no problem :) updated now.